### DISCUSSION TITLE
DataArray.sort_values()

### DISCUSSION THREAD

**Discussion by raybellwaves** [pydata/xarray#5795] [Ideas] [answered]
When going through https://github.com/raybellwaves/PyData-Portland-2021-talk/blob/main/talk.ipynb someone asked about the syntax of `da_1d.sortby(da_1d)`. It may help to have a `DataArray.sort_values` method which works on 1d arrays which wraps  `da_1d.sortby(da_1d)`

Adding will be helpful for pandas users but does duplicate an existing method/API

**Comment by max-sixty** [comment, marked as answer]
I agree that `da_1d.sortby(da_1d)` could be simpler. Some options:
- `da_1d.sortby()` could work for 1d arrays? Generally we try to not special-case by dimension number though
- `da_1d.sortby(lambda x: x)` could be a nice general solution (it's already possible to pass lambdas to methods like `where`), though maybe not that friendly to newcomers.

I'm not sure it's worth another method — as you point out — it duplicates another method, and could make it confusing on which to use.

What are your thoughts @raybellwaves ?

**Reply by raybellwaves** [reply]
I think it's fine as it. Part of my posting is in learning the difference API across the ecosystem. e.g. in numpy it's `sort` https://numpy.org/doc/stable/reference/generated/numpy.sort.html and pandas it's `sort_values` https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sort_values.html and `sort_index` https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sort_index.html. I think xarray's `sortby` encompasses these. There's a page in the docs for this as well: http://xarray.pydata.org/en/stable/user-guide/reshaping.html#sort

**Reply by dcherian** [reply]
You could add an entry to https://xarray.pydata.org/en/latest/howdoi.html maybe that helps. And also a docstring example.
